src/Tools/CodeGenerator/Featurizers.yaml

```diff
@@ -1557,10 +1557,10 @@ featurizers:
       The resulting matrix size is K rows x N cols, where K is the number of offsets and N is the horizon.
 
       Horizon and offsets should be passed in during construction. Offsets are passed in as a vector of ints so multiple lag orders can be applied within one featurizer call.
-      Output type is a matrix of double where rows are grouped by different offsets and columns are grouped by horizon.
+      Output type is a matrix of optional<T> where rows are grouped by different offsets and columns are grouped by horizon.
 
       C++-style pseudo signature:
-        template <typename T> matrix<double> execute(std::vector<std::string> const &, T const &> const &value);
+        template <typename T> matrix<T?> execute(std::vector<std::string> const &, T const &> const &value);

```

Comment:
tuple?

Reply:
The return type should just be a matrix<nonstd::optional<T>>

Reply:
I mean the input type:
std::vector<std::string> const &, T const &>
looks missing tuple?

Reply:
Oh! That is correct. Added tuple.